src/odemis/acq/move.py

```diff
@@ -0,0 +1,317 @@
+    """
+    Computes the position on the path between LOADING and IMAGING.
+    If it’s too far from the line between LOADING → IMAGING, then it’s considered out of the path.
+    :param current_pos: (dict str->float) Current position of the stage
+    :param start_pos: (dict str->float) Either LOADING or IMAGING position to start the movement from
+    :param end_pos: (dict str->float)Either LOADING or IMAGING position to end the movement to
+    :return:(0<=float<=1, or None) Ratio of the progress, None if it's far away from of the path
+    """
+
+    def get_distance(start, end):
+        # Calculate the euclidean distance between two 3D points
+        sp = numpy.array([start['x'], start['y'], start['z']])
+        ep = numpy.array([end['x'], end['y'], end['z']])
+        return scipy.spatial.distance.euclidean(ep, sp)
+
+    def check_axes(pos):
+        if not {'x', 'y', 'z'}.issubset(set(pos.keys())):
+            raise ValueError("Missing x,y,z axes in {} for correct distance measurement.".format(pos))
+
+    # Check we have the x,y,z axes in all points
+    check_axes(current_pos)
+    check_axes(start_pos)
+    check_axes(end_pos)
+    # Get distance for current point in respect to start and end
+    from_start = get_distance(start_pos, current_pos)
+    to_end = get_distance(current_pos, end_pos)
+    total_length = get_distance(start_pos, end_pos)
+    # Check if current position is on the line from start to end position
+    # That would happen if start_to_current +  current_to_start = total_distance from start to end
+    if util.almost_equal((from_start + to_end), total_length, rtol=RTOL_PROGRESS):
+        return min(from_start / total_length, 1.0)  # Clip in case from_start slightly > total_length
+    else:
+        return None
+
+
+def _isInRange(stage, axis):
+    """
+    Check if current position is within active range
+    :return: True if position in active range, False otherwise
+    """
+    if axis not in stage.axes.keys():
+        raise ValueError("Axis %s is not found in stage axes", axis)
+    pos = stage.position.value[axis]
+    active_range = [r for r in stage.getMetadata()[model.MD_POS_ACTIVE_RANGE][axis]]
+    # Add 1% margin for hardware slight errors
+    margin = (active_range[1] - active_range[0]) * 0.01
+    return (active_range[0] - margin) <= pos <= (active_range[1] + margin)
+
+
+def _isNearDeactive(current_pos, stage_deactive, axis):
+    """
+    Check whether given axis is near stage's FAV_POS_DEACTIVE position
+    :return: True if the axis is near (within 1 mm or 0.1 rad) FAV_POS_DEACTIVE, False otherwise
+    :raises ValueError if axis is unknown
+    """
+    current_value = current_pos[axis]
+    deactive_value = stage_deactive[axis]
+    if axis in {'x', 'y', 'z'}:
+        return abs(deactive_value - current_value) < ATOL_LINEAR_POS
+    elif axis in {'rx', 'rz'}:
+        return util.rot_almost_equal(current_value, deactive_value, atol=ATOL_ROTATION_POS)
```

Comment:
Thanks for clarifying this, added `rot_almost_equal` can you please check? @pieleric